modules/javafx.graphics/src/main/java/javafx/css/Stylesheet.java

```diff
@@ -278,25 +279,48 @@ final void readBinary(int bssVersion, DataInputStream is, String[] strings)
      * css version or if an I/O error occurs while reading from the stream
      */
     public static Stylesheet loadBinary(URL url) throws IOException {
+        if (url == null) {
+            return null;
+        }
+
+        try (InputStream stream = url.openStream()) {
+            return loadBinary(stream, url.toExternalForm());
+        } catch (FileNotFoundException ex) {
+            return null;
+        }
+    }
 
-        if (url == null) return null;
+    /**
+     * Loads a binary stylesheet from a stream.
+     *
+     * @param stream the input stream
+     * @return the loaded {@code Stylesheet}
+     * @throws IOException if the binary stream corresponds to a more recent binary
+     * css version or if an I/O error occurs while reading from the stream
+     *
+     * @since 17
+     */
+    public static Stylesheet loadBinary(InputStream stream) throws IOException {
+        return loadBinary(stream, null);
+    }
 
+    private static Stylesheet loadBinary(InputStream stream, String uri) throws IOException {
         Stylesheet stylesheet = null;
 
         try (DataInputStream dataInputStream =
-                     new DataInputStream(new BufferedInputStream(url.openStream(), 40 * 1024))) {
+                     new DataInputStream(new BufferedInputStream(stream, 40 * 1024))) {
 
             // read file version
             final int bssVersion = dataInputStream.readShort();
             if (bssVersion > Stylesheet.BINARY_CSS_VERSION) {
-                throw new IOException(url.toString() + " wrong binary CSS version: "
+                throw new IOException("Wrong binary CSS version: "
```

Comment:
If the `url` string is non-null shouldn't it still be part of the exception message?

Reply:
True.